metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_queries.py

```diff
@@ -538,6 +538,38 @@ def parse_ddl_query(
             self.report.num_ddl_queries_dropped += 1
             return None
 
+    def _get_query_type(self, query_type: str, query_text: Optional[str] = None) -> str:
+        if query_type == "CREATE_TABLE" and query_text and "CLONE" in query_text.upper():
+            return "CLONE"
+        return query_type
+
+    def process_query_event(self, event: Dict[str, Any]) -> Optional[PreparsedQuery]:
```

Comment:
Have you considered a more precise name? For example,

`generate_clone_lineage_from_query_event_if_present`?